pkg/client/redhatsso/client.go

```diff
@@ -0,0 +1,154 @@
+package redhatsso
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	serviceaccountsclient "github.com/redhat-developer/app-services-sdk-go/serviceaccounts/apiv1internal/client"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type SSOClient interface {
+	GetToken() (string, error)
+	GetServiceAccounts(accessToken string, first int, max int) ([]serviceaccountsclient.ServiceAccountData, error)
+	GetServiceAccount(clientId string, accessToken string) (serviceaccountsclient.ServiceAccountData, error)
+	CreateServiceAccount(accessToken string, name string, description string) (serviceaccountsclient.ServiceAccountData, error)
+	DeleteServiceAccount(clientId string, accessToken string) error
+	UpdateServiceAccount(clientId string, name string, description string, accessToken string) (serviceaccountsclient.ServiceAccountData, error)
+	RegenerateClientSecret(accessToken string, id string) (serviceaccountsclient.ServiceAccountData, error)
+}
+
+func NewSSOClient(config *RedhatSSOConfig) SSOClient {
+	return &rhSSOClient{
+		config: config,
+	}
+}
+
+var _ SSOClient = &rhSSOClient{}
+
+type rhSSOClient struct {
+	config        *RedhatSSOConfig
+	configuration *serviceaccountsclient.Configuration
+}
+
+type tokenResponse struct {
+	AccessToken      string `json:"access_token"`
+	ExpiresIn        int    `json:"expires_in"`
+	RefreshExpiresIn int    `json:"refresh_expires_in"`
+	TokenType        string `json:"token_type"`
+	NotBeforePolicy  int    `json:"not-before-policy"`
+	Scope            string `json:"scope"`
+}
+
+func (kc *rhSSOClient) getConfiguration(accessToken string) *serviceaccountsclient.Configuration {
+	if kc.configuration == nil {
+		kc.configuration = &serviceaccountsclient.Configuration{
+			DefaultHeader: map[string]string{
+				"Authorization": fmt.Sprintf("Bearer %s", accessToken),
+				"Content-Type":  "application/json",
+			},
+			UserAgent: "OpenAPI-Generator/1.0.0/go",
+			Debug:     false,
+			Servers: serviceaccountsclient.ServerConfigurations{
+				{
+					URL: kc.config.KafkaRealm.APIEndpointURI,
+				},
+			},
+		}
+	}
+	return kc.configuration
+}
+
+func (kc *rhSSOClient) GetToken() (string, error) {
```

Comment:
Can we  also add cache for token something like: GetCachedToken
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/keycloak/client.go#L178

Reply:
done

Reply:
done